code style guides created

synthmeat opened the pull request:
I've created JSCS configuration file to maximally conform to whatever's already in the codebase. Only thing I couldn't shove in was (part of) alignments on array literals (only 2 instances of those are in project). ```templates``` can't be stylized either without significant workarounds. Client is not stylized either.

There's no enforcement of the style in any way (say, grunt jscs task).

Also to note, it was all fairly consistent to begin with.

evantahler commented:
So before I merge this in, I guess it would be best to also publish a guide on how to use these hint files... with guides for every OS and a few editors (Atom and Sublime Text are must-haves).

synthmeat commented:
Sure, I'll whip up something over the weekend. ``Contributing``` chapter in docs good entry point?

evantahler commented:
@synthmeat yes please! We can also make a [`CONTRIBUTING`](https://github.com/blog/1184-contributing-guidelines) file in the project as well.

Note: looks like you need to rebase

neilstuartcraig commented:
Hi all

I heard about this today:
https://github.com/kentcdodds/all-contributors

I haven't looked in detail but might be worth considering. 

Cheers

evantahler commented:
@neilstuartcraig I think that's not relevant here.  Open a new issue with it?

neilstuartcraig commented:
Ah sorry, i was looking at it in terms of the `CONTRIBUTING` file - sorry, probably me cluttering the issue.

synthmeat commented:
Moved from JSHint + JSCS to just ESLint after evaluating all of them for my own projects. Single tool, has option to mark things as warnings as well as errors, and (what I particulary like) does not do anything  by default.

evantahler commented on bin/methods/actions.js:
```diff
@@ -1,7 +1,7 @@
-  var ActionheroPrototype = require(binary.actionheroRoot + '/actionhero.js').actionheroPrototype;
-  var actionhero = new ActionheroPrototype();
-  var configChanges = { logger: {transports: null} };
+  var actionheroPrototype = require(binary.actionheroRoot + '/actionhero.js').actionheroPrototype;
```
I thought constructors were always supposed to be capitalized?

evantahler commented on config/api.js:
```diff
@@ -34,15 +34,15 @@ exports.default = {
-        'server':      [ __dirname + '/../servers'      ] ,
-        'initializer': [ __dirname + '/../initializers' ] ,
-        'plugin':      [ __dirname + '/../node_modules' ] ,
-        'locale':      [ __dirname + '/../locales'      ]
```
I assume we initially did this kind of thing for legibility, but I'm OK with the change.

evantahler replied:
Asked some people, they hate whitespace... remove it!

evantahler commented on initializers/config.js:
```diff
@@ -76,7 +76,9 @@ module.exports = {
           addConfigPath(pathToCheck.split(','), true);
         }
         else {
-          if (pathToCheck.charAt(0) !== '/') pathToCheck = path.resolve(api.projectRoot, pathToCheck);
```
Thanks! I personally don't like this format

evantahler commented on package.json:
```diff
@@ -52,7 +52,8 @@
     "coveralls": "^2.11.4",
-    "istanbul": "^0.4.0"
+    "istanbul": "^0.4.0",
+    "eslint": "^2.3.0"
```
What does requiring this do?  It's never loaded in the project anywhere...

evantahler commented:
A few questions added... looking great!

synthmeat commented:
`eslint` is added as dev dep because that way we get control over its version, and plugins (as far as I've seen) have preference for project's own `eslint` binary.

I've actually now dropped requirement for upper-casing constructors since there's quite a few of them to be added exceptions just from `node-resque`. I can work that out on both `actionhero` and `node-resque` some time later?

`.eslintrc` was actually broken in prior commit (so it didn't report some issues I've fixed now). Fixed it and ran through everything once again. Just to confirm - we'd like this style:
```javascript
function(){
  try{
    if(stuff){
      stuff();
    }else if{
      stuff();
    }else{
     stuff();
    }
  }catch(e){
    stuff();
  }
};
```
?


Lastly, meanwhile there has been a lot of reserved keywords introduced as properties (`default`, `class`, `do`) so I've wrapped those to be name-accessed.

evantahler commented:
Ok, I think the only real issue that might upset people is `thing.default` => `thing['default']`.  Does anyone else have an opinion on this?

This is a heroic effort, @synthmeat, and I want to merge it in!

evantahler commented:
fuck it, this is awesome... and should be merged in ASAP.

synthmeat commented:
🎉
